source/extensions/transport_sockets/tls/ssl_socket.cc

```diff
@@ -122,17 +119,16 @@ Network::IoResult SslSocket::doRead(Buffer::Instance& read_buffer) {
   PostIoAction action = PostIoAction::KeepOpen;
   uint64_t bytes_read = 0;
   while (keep_reading) {
+    uint64_t bytes_read_this_iteration = 0;
     // We use 2 slices here so that we can use the remainder of an existing buffer chain element
     // if there is extra space. 16K read is arbitrary and can be tuned later.
-    Buffer::RawSlice slices[2];
-    uint64_t slices_to_commit = 0;
-    uint64_t num_slices = read_buffer.reserve(16384, slices, 2);
-    for (uint64_t i = 0; i < num_slices; i++) {
-      auto result = sslReadIntoSlice(slices[i]);
-      if (result.commit_slice_) {
-        slices_to_commit++;
-        bytes_read += slices[i].len_;
-      }
+    Buffer::Reservation reservation = read_buffer.reserve(16384);
```

Comment:
I think it would need some kind of heuristic, similar to what I'm trying to do in `maybeLinearize()` on the other PR. If there's 1 byte left in the last slice, it's better to allocate a new slice.

Reply:
It may depend on what we'ld like to optimize for.  Returning that 1 byte would help with packing.  That said, an API like: Buffer::RawSlice reserveRange(min, max)
could be the best of both worlds.